openstack_dashboard/test/selenium/integration/test_user_credentials.py

```
@@ -0,0 +1,175 @@
+    ))
+    driver.get(url)
+    credentials_sdk_before = list(openstack_admin.identity.credentials())
+    assert(len(credentials_sdk_before) == 0)
+    driver.find_element_by_link_text("Create User Credential").click()
+    user_credential_form = driver.find_element_by_css_selector(
+        ".modal-dialog form")
+    widgets.select_from_specific_dropdown_in_form(
+        user_credential_form, 'id_cred_type', 'TOTP')
+    user_credential_form.find_element_by_css_selector(".btn-primary").click()
+    messages = widgets.get_and_dismiss_messages(driver)
+    assert f'Success: User credential created successfully.' in messages
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)